### PR TITLE
Show a list of installed plugins for Jetpack sites

### PR DESCRIPTION
**Pull request by koke**
First part of #7471:

- Feature flag: plugins only enabled on developer builds for now.
- Blog support flag: plugins only enabled for Jetpack sites via REST.
- UI: a basic plugin list with name, state, and a placeholder icon.
- Backend: Implemented remote and unit tests to load plugin list from the API.

simulator screen shot 8 aug 2017 11 12 23

To test:

- For a Jetpack site, go to Site > Plugins
- Verify that it list the currently installed plugins

Needs review: @elibud

### CODE REVIEW

File: WordPress/Classes/ViewRelated/Plugins/PluginListViewController.swift [hunk 1]
```diff
@@ -0,0 +1,79 @@
+import UIKit
+import WordPressKit
+
+class PluginListViewController: UITableViewController, ImmuTablePresenter {
+    fileprivate lazy var handler: ImmuTableViewHandler = {
+        return ImmuTableViewHandler(takeOver: self)
+    }()
```
Comment (elibud) [inline WordPress/Classes/ViewRelated/Plugins/PluginListViewController.swift]: Missing new line after this one?

File: WordPress/Classes/ViewRelated/Plugins/PluginListViewController.swift [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        guard let api = blog.wordPressComRestApi(),
+            let service = PluginServiceRemote(wordPressComRestApi: api) else {
+            return nil
+        }
+
+        self.init(siteID: Int(blog.dotComID!), service: service)
+    }
+
+    let siteID: Int
+    let service: PluginServiceRemote
```
Comment (elibud) [inline WordPress/Classes/ViewRelated/Plugins/PluginListViewController.swift]: Personally I would add a new line after this line and I would move the convenience init to after the init below this comment, but that may be me just being particular about ordering things.

File: WordPressKit/WordPressKit/PluginState.swift [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    public let name: String
+    public let version: String?
+    public let autoupdate: Bool
+}
+
+public extension PluginState {
+    var stateDescription: String {
+        switch (active, autoupdate) {
+        case (false, _):
+            return NSLocalizedString("Inactive", comment: "A plugin is not active on the site")
```
Comment (elibud) [inline WordPressKit/WordPressKit/PluginState.swift]: Instead of "A plugin"->"The plugin" ?

Comment (elibud) [inline WordPressKit/WordPressKit/PluginState.swift]: We could also add a comment that it should be the same string as in WP.com

**Comment by elibud** [comment]
One more question, in the description it says:

> a basic plugin list with name, version, and a placeholder icon.

But we are not currently showing the version but the state, right?

**Comment by koke** [comment]
> we are not currently showing the version but the state

Correct, I changed that in the last commit after seeing the new mockups. I updated the screenshot but missed the description

**Comment by koke** [comment]
Ready for another look

**Comment by koke** [comment]
🎉 Thanks for the review!

**Pull request merged**